packages/flutter/lib/src/rendering/child_view.dart

```diff
@@ -0,0 +1,207 @@
+
+import 'box.dart';
+import 'object.dart';
+
+// TODO(abarth): Populate the view host.
+final mojom.ViewHost _viewHost = null;
+
+class ChildViewConnection {
+  ChildViewConnection({ this.url }) {
+    mojom.ServiceProviderProxy incommingServices = new mojom.ServiceProviderProxy.unbound();
+    mojom.ServiceProviderStub outgoingServices = new mojom.ServiceProviderStub.unbound();
+    _connection = new ApplicationConnection(outgoingServices, incommingServices);
+    _unresolvedViewToken = _connectToView(incommingServices, outgoingServices);
+  }
+
+  final String url;
+
+  ApplicationConnection get connection => _connection;
+  ApplicationConnection _connection;
+
+  Future<mojom.ViewToken> _unresolvedViewToken;
+  mojom.ViewToken _viewToken;
+
+  Future<mojom.ViewToken> _connectToView(
+      mojom.ServiceProviderProxy incommingServices,
+      mojom.ServiceProviderStub outgoingServices) async {
+    assert(_viewToken == null);
+    mojom.ViewProviderProxy viewProvider = new mojom.ViewProviderProxy.unbound();
+    shell.connectToService(url, proxy);
+    mojom.ViewToken viewToken = (await viewProvider.ptr.createView(incommingServices, outgoingServices)).viewToken;
+    viewProvider.close();
+    assert(_viewToken == null);
+    _viewToken = viewToken;
+    assert(_viewKey == null);
+    if (_attached)
+      _addChildToViewHost();
+    return viewToken;
+  }
+
+  static int _nextViewKey = 1;
+  int _viewKey;
+
+  void _addChildToViewHost() {
+    assert(_attached);
+    assert(_viewToken != null);
+    assert(_viewKey == null);
+    _viewKey = _nextViewKey++;
+    _viewHost.addChild(_viewKey, _viewToken);
+  }
+
+  void _removeChildFromViewHost() {
+    assert(!_attached);
+    assert(_viewToken != null);
+    assert(_viewKey != null);
+    _viewHost.removeChild(_viewKey);
+    _viewKey = null;
+  }
+
+  // The number of render objects attached to this view. In between frames, we
+  // might have more than one connected if we get added to a new render object
+  // before we get removed from the old render object. By the time we get around
+  // to computing our layout, we must be back to just having one render object.
+  int _attachments = 0;
+  bool get _attached => _attachments > 0;
+
+  void _attach() {
+    assert(_attachments >= 0);
+    ++_attachments;
+    if (_viewToken != null && _viewKey == null)
+      _addChildToViewHost();
+  }
+
+  void _detach() {
+    assert(_attached);
+    --_attachments;
+    scheduleMicrotask(_removeChildFromViewHostIfNeeded);
+  }
+
+  void _removeChildFromViewHostIfNeeded() {
+    assert(_attachments >= 0);
+    if (_attachments == 0)
+      _removeChildFromViewHost();
+  }
+
+  Future<mojom.ViewLayoutInfo> _layout({ Size size, double scale }) async {
+    assert(_attached);
+    assert(_attachments == 1);
+    assert(_viewKey != null);
+    int width = (size.width * scale).round();
+    int height = (size.height * scale).round();
+    mojom.BoxConstraints childConstraints = new mojom.BoxConstraints()
```

Comment:
We'll want to do that eventually.  The tricky part is that we need to compute a size for the render object synchronously.  If the child comes back with a different size, we'll need to trigger another layout, which could cause another child view to change its size, etc.  The whole thing could cascade.

My goal is to get to the point where we can see things on screen and then we can iterate from there.